src/Application/Executor.cs

```diff
@@ -1,95 +1,48 @@
-﻿using System;
-using System.Diagnostics;
-using System.IO;
-using System.Runtime.InteropServices;
+﻿using System.Runtime.InteropServices;
 using System.Threading.Tasks;
 using Symbolica.Application.Collection;
 using Symbolica.Application.Computation;
 using Symbolica.Application.Implementation;
 using Symbolica.Computation;
 using Symbolica.Deserialization;
-using Symbolica.Expression;
 using Symbolica.Implementation;
 using Symbolica.Implementation.System;
 using Symbolica.Representation;
 
 namespace Symbolica.Application
 {
-    internal static class Executor
+    internal sealed class Executor
     {
-        public static async Task<Result> Run(string directory, Options options)
-        {
-            var buildImage = Environment.GetEnvironmentVariable("SYMBOLICA_BUILD_IMAGE");
-            var translateImage = Environment.GetEnvironmentVariable("SYMBOLICA_TRANSLATE_IMAGE");
-
-            File.Delete(Path.Combine(directory, "symbolica.bc"));
-            File.Delete(Path.Combine(directory, ".symbolica.bc"));
+        private readonly Options _options;
 
-            await CallExternalProcess(directory, buildImage == null
-                ? "./symbolica.sh"
-                : $"docker run -v $(pwd):/code {buildImage}");
+        public Executor(Options options)
```

Comment:
I'd have probably kept this as a `static` class with a `static` method, but I bet you knew I'd say that anyway. 😉

Reply:
I was going to make a similar change on the Hosted side, since we don't really need the static entry point any more, but I am scared to now.